public/app/plugins/datasource/loki/datasource.ts

```diff
@@ -754,6 +778,26 @@ export class LokiDatasource
       return addLabelToQuery(queryExpr, key, value, operator, true);
     }
   }
+
+  // Used when running queries through backend
+  filterQuery(query: LokiQuery): boolean {
+    if (query.hide || query.expr === '') {
+      return false;
+    }
+    return true;
+  }
+
+  // Used when running queries through backend
+  applyTemplateVariables(target: LokiQuery, scopedVars: ScopedVars): Record<string, any> {
+    // We want to interpolate these variables on backend
```

Comment:
Do we want to do it though? Because we don't have step parameter for Loki (yet - and I am not sure if/when we'll add it). Therefore we could just use the calculated interval in the same way as it is used in other data sources.

Reply:
hmm. i hope i'm not misunderstanding this 😄 

so, i definitively do not want to calculate the `interval` on the backend. i'm just using what gets sent to me ( https://github.com/grafana/grafana/blob/main/pkg/tsdb/loki/parse_query.go#L72 )

the thing is, for alerting-queries, i need to do the string-replacing of `$__interval` in the query-string. so i thought it would make sense to do it the same way for from-frontend-coming queries too. we could just let the frontend interpolate them here, the value should be equal, but maybe some javascript-string-formatting is different from go-string-formatting... it felt "more correct" to do both at the same place.

(it's like this in influxdb flux-mode too at https://github.com/grafana/grafana/blob/main/public/app/plugins/datasource/influxdb/datasource.ts#L191-L192)

does this make sense?

Reply:
Yes, makes sense! Thanks for context and explanation.